gateway-js/src/core/__tests__/core.test.ts

```diff
@@ -0,0 +1,285 @@
+import { ApolloGateway } from '../../..';
+import { graphqlErrorSerializer } from 'apollo-federation-integration-testsuite';
+
+expect.addSnapshotSerializer(graphqlErrorSerializer);
+
+describe('core v0.1', () => {
+  it("doesn't throw errors / ignores 'for' args", () => {
```

Comment:
I'm going with option 1 for now since we can relax this in the future and the machinery for erroring currently exists while warning doesn't. I do think option 2 is pretty reasonable though, so not opposed to us making that change as a fast-follow.
77d1570